server/src/main/java/org/elasticsearch/common/util/BigByteArray.java

```diff
@@ -127,6 +127,16 @@ public void fill(long fromIndex, long toIndex, byte value) {
         }
     }
 
+    @Override
+    public boolean hasArray() {
+        return false;
+    }
+
+    @Override
+    public byte[] array() {
+        throw new UnsupportedOperationException();
```

Comment:
`assert false` here, we shouldn't be calling this?

Reply:
++